jplag.frontend-utils/src/main/java/de/jplag/LanguageLoader.java

```diff
@@ -0,0 +1,79 @@
+package de.jplag;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.TreeMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class contains methods to load {@link Language Languages}.
+ * @author Dominik Fuchss
+ */
+public final class LanguageLoader {
+    private static final Logger logger = LoggerFactory.getLogger(LanguageLoader.class);
+
+    private static Map<String, Language> loaded = null;
+
+    private LanguageLoader() {
+        throw new IllegalAccessError();
+    }
+
+    /**
+     * Load all languages that are currently in the classpath.
+     * @return the languages
+     */
+    public static synchronized List<Language> getAllAvailableLanguages() {
```

Comment:
This could also be a `Set`. Sorting alphabetically is not done anymore here and has no semantic meaning to the language provider. Rather, it should be done directly in the CLI.

Reply:
Language would have to implement equals/hashcode. Therefore I use a list here :)